dislib/data/classes.py

```diff
@@ -58,6 +58,8 @@ def subset_size(self, index):
         if self._sizes[index] is None:
             size = compss_wait_on(_subset_size(self._subsets[index]))
             self._sizes[index] = size
+        elif type(self._sizes[index]) == Future:
+            self._sizes = compss_wait_on(self._sizes)
```

Comment:
why are you synchronizing all the _sizes vector instead of just _sizes[index] ?

Reply:
to avoid multiple synchronizations

Reply:
is it necessary? I mean, does calling compss_wait_on have overhead if it's not a future? because otherwise it would be equivalent to:
```
if self._sizes[index] is None:
￼              size = compss_wait_on(_subset_size(self._subsets[index]))		￼              
￼              self._sizes[index] = size	
self._sizes = compss_wait_on(self._sizes)

```

Reply:
I know, but I don't know the cost of running compss_wait_on, and you risk performing an uneeded synchronization